test/Integration/IntegrationTestBase.cs

```diff
@@ -164,6 +166,59 @@ private void ExecuteAllScriptsInFolder(string folder)
             }
         }
 
+        /// <summary>
+        /// Run `mvn clean package` to build the Java function app.
+        /// </summary>
+        private void BuildJavaFunctionApp(string workingDirectory)
+        {
+            this.Mvn = new Process()
+            {
+                StartInfo = new ProcessStartInfo
+                {
+                    FileName = GetMavenPath(),
+                    Arguments = "clean package",
+                    WorkingDirectory = workingDirectory,
+                    WindowStyle = ProcessWindowStyle.Hidden,
+                    RedirectStandardOutput = true,
+                    RedirectStandardError = true,
+                    UseShellExecute = false
+                }
+            };
+
+            var taskCompletionSource = new TaskCompletionSource<bool>();
+            this.Mvn.OutputDataReceived += SignalStartupHandler;
+
+            this.Mvn.Start();
+            this.Mvn.OutputDataReceived += this.GetTestOutputHandler();
+            this.Mvn.ErrorDataReceived += this.GetTestOutputHandler();
+            this.Mvn.BeginOutputReadLine();
+            this.Mvn.BeginErrorReadLine();
+
+            this.LogOutput($"Building Java function app...");
+
+            const int buildJavaAppTimeoutInSeconds = 60;
+            bool isCompleted = taskCompletionSource.Task.Wait(TimeSpan.FromSeconds(buildJavaAppTimeoutInSeconds));
+            Assert.True(isCompleted, "Java function app did not build successfully");
+
+            this.LogOutput("Java build successful!");
+            this.Mvn.OutputDataReceived -= SignalStartupHandler;
+
+            void SignalStartupHandler(object sender, DataReceivedEventArgs e)
```

Comment:
I'd suggest moving this up before you attach it - it's confusing to have it declared after it's used